libraries/AC_AutoTune/AC_AutoTune.h

```diff
@@ -215,14 +324,120 @@ class AC_AutoTune {
+    // generic method used by subclasses to update gains for the rate ff up tune type
+    virtual void updating_rate_ff_up_all(AxisType test_axis)=0;
+    // generic method used by subclasses to update gains for the rate ff down tune type
+    virtual void updating_rate_ff_down_all(AxisType test_axis)=0;
+    // generic method used by subclasses to update gains for the max gain tune type
+    virtual void updating_max_gains_all(AxisType test_axis)=0;
+
+    // Feedforward test used to determine Rate FF gain
+    void rate_ff_test_init();
+    void rate_ff_test_run(float max_angle_cds, float target_rate_cds, float dir_sign);
+
+    // dwell test used to perform frequency dwells for rate gains
+    void dwell_test_init(float filt_freq);
+    void dwell_test_run(uint8_t freq_resp_input, float start_frq, float stop_frq, float &dwell_gain, float &dwell_phase);
+
+    // dwell test used to perform frequency dwells for angle gains
+    void angle_dwell_test_init(float filt_freq);
+    void angle_dwell_test_run(float start_frq, float stop_frq, float &dwell_gain, float &dwell_phase);
+
+    float waveform(float time, float time_record, float waveform_magnitude, float wMin, float wMax);
+
+    uint8_t  ff_test_phase;                         // phase of feedforward test
+    float    test_command_filt;                     // filtered commanded output
+    float    test_rate_filt;                        // filtered rate output
+    float    command_out;
+    float    test_tgt_rate_filt;                    // filtered target rate
+    float    filt_target_rate;
+    bool     ff_up_first_iter;                      //true on first iteration of ff up testing
+    float    test_gain[20];                             // gain of output to input
+    float    test_freq[20];
+    float    test_phase[20];
+    float    dwell_start_time_ms;
+    uint8_t  freq_cnt;
+    uint8_t  freq_cnt_max;
+    float    curr_test_freq;
+    float    curr_test_gain;
+    float    curr_test_phase;
+    Vector3f start_angles;
+    uint32_t settle_time;
+    uint32_t phase_out_time;
+    float    waveform_freq_rads;  //current frequency for chirp waveform
+    float    start_freq;  //start freq for dwell test
+    float    stop_freq;   //ending freq for dwell test
+    float    trim_pff_out;  // trim output of the PID rate controller for P, I and FF terms
+    float    trim_meas_rate;  // trim measured gyro rate
+
+    //variables from rate FF test
+    float trim_command_reading;
+    float trim_heading;
+    float rate_request_cds;
+    float angle_request_cd;
+
+    // variables from rate dwell test
+    Vector3f trim_attitude_cd;
+    Vector3f filt_attitude_cd;
+    Vector2f filt_att_fdbk_from_velxy_cd;
+    float filt_command_reading;
+    float filt_gyro_reading;
+    float filt_tgt_rate_reading;
+    float trim_command;
+
+    // variables from angle dwell test
+    float trim_yaw_tgt_reading;
+    float trim_yaw_heading_reading;
+//    Vector2f filt_att_fdbk_from_velxy_cd;
```

Comment:
probably best to remove this commented out code..

Reply:
done